ports/nrf/common-hal/busio/UART.c

```diff
@@ -237,6 +266,10 @@ size_t common_hal_busio_uart_read(busio_uart_obj_t *self, uint8_t *data, size_t
 
     uint64_t start_ticks = supervisor_ticks_ms64();
 
+    if (len > ringbuf_capacity(&self->ringbuf)) {
+        mp_raise_ValueError(translate("Reading >receiver_buffer_size bytes is not supported"));
+    }
```

Comment:
I don't think this check is worth it. This is called through the stream api with len 1 almost always. Removing it will reduce the build size for the whole error string.

Reply:
E.g. `msgpack` will timeout if the buffer is less than 256 bytes.

As a compromise I've commented out the check.